### ISSUE TITLE
Missing DATA XREFS for cjmp ops

### ISSUE THREAD

**Issue by astuder** [radare/radare2#10066]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Debian x86 64
| File format of the file you reverse (mandatory)      | binary dump
| Architecture/bits of the file (mandatory)            | 8051
| r2 -v full output, **not truncated** (mandatory)     | radare2 2.6.0-git 18032 @ linux-x86-64 git.1.5.0-3175-g5d11df9c1
commit: 5d11df9c163873420d89f78e75ee18c4b0e31683 build: 2018-05-10__07:18:41

### Expected behavior

DATA XREFS for memory locations referenced in conditional jump.

### Actual behavior

No DATA XREFS created for memory locations referenced in conditional jumps.

### Steps to reproduce the behavior 

The example is for 8051. 

Conditional jump depending on the value at a memory address:

```
[0x0000c687]> pd 1 @0xc69a
        ┌─< 0x0000c69a      307415         jnb 0x2e.4, 0xc6b2          ; [0x542e:1]=0
[0x0000c687]> ao 1 @0xc69a
address: 0xc69a
opcode: jnb 0x2e.4, 0xc6b2
mnemonic: jnb
prefix: 0
id: 0
bytes: 307415
ptr: 0x0000542e
refptr: 1
size: 3
type: cjmp
esil: 4,_idata,46,+,[1],>>,1,&,!,?{,50866,pc,=,}
jump: 0x0000c6b2
fail: 0x0000c69d
stack: null
cond: al
family: cpu
stackop: null
```

DATA XREF for 0x0000c687 is missing:

```
[0x0000c687]> pd 1 @0x542e
            ; XREFS: DATA 0x0000ff67  DATA 0x00008504  DATA 0x00008766  DATA 0x0000845c  
            ; XREFS: DATA 0x00008248  DATA 0x00008377  DATA 0x00008c93  DATA 0x0000a815  
            ; XREFS: DATA 0x0000a813  DATA 0x0000a837  DATA 0x0000fdd8  DATA 0x000083f6  
            ; XREFS: DATA 0x0000a82e  
            0x0000542e      .byte 0x00
```

The example uses 8051 bit addressing, which the 8051 anal translates to a "virtual" memory address. But I don't think that's the issue as DATA XREFS are created for mov:

```
[0x0000a9c1]> pd 1
            0x0000a9c1      9229           mov 0x25.1, c               ; var.flags_25 ; [0x5425:1]=0 ; low power flag
[0x0000a9c1]> ao 1
address: 0xa9c1
opcode: mov 0x25.1, c
mnemonic: mov
prefix: 0
id: 0
bytes: 9229
ptr: 0x00005425
refptr: 1
size: 2
type: mov
esil: 1,c,<<,1,1,<<,255,^,_idata,37,+,[1],&,|,_idata,37,+,=[1],
stack: null
family: cpu
stackop: null
[0x0000a9c1]> pd 1 @0x5425
            ;-- var.flags_25:
            ; XREFS: DATA 0x0000d22a  DATA 0x0000fc22  DATA 0x000090c1  DATA 0x0000ff59  
            ; XREFS: DATA 0x000083ab  DATA 0x00009899  DATA 0x00009088  DATA 0x0000b8d3  
            ; XREFS: DATA 0x0000bd69  DATA 0x0000cf78  DATA 0x0000bd5d  DATA 0x0000cf90  
            ; XREFS: DATA 0x0000a132  DATA 0x0000dbcf  DATA 0x0000a9c1  DATA 0x0000d1fc  
            ; XREFS: DATA 0x0000a78f  DATA 0x0000a985  DATA 0x0000a956  DATA 0x0000a1e4  
            ; XREFS: DATA 0x0000a7f7  DATA 0x000081d5  
            0x00005425      .byte 0x00                                 ; 1:low power

```

**Comment by radare** [comment]
A conditional jump dont references code

**Comment by radare** [comment]
Is this code from git? This stuff changed 2 days ago

**Comment by astuder** [comment]
Yes, pulled git today. I don't think it's related to the recent rework of references, which works great (solved a few other issues).

Conditional jump, with a condition that accesses memory. CODE reference is created. DATA reference is missing. Unless there's another op-type I should use for these kind of jumps.

**Comment by radare** [comment]
Ok

**Comment by XVilka** [comment]
Check also the "anal.ijmp" option.

**Comment by astuder** [comment]
Digging a bit deeper: I use aar command to generate xrefs. When I call afr on the problematic address (afr @0xc96a), the missing xref is created.

There are (at least) two implementations of xref analysis:
r_core_anal_search_xrefs in core/canal.c, invoked by aar
fcn_recurse in anal/fcn.c, invoked by afr

fcn_recurse looks much more comprehensive.

Consolidate both into an r_anal_op_xref?

**Comment by radare** [comment]
Well i think is good to have both. And one of them is part of the search logic. How do you plan to merge this?

**Comment by astuder** [comment]
I thought about moving the logic that creates the xrefs for the current op into a function, then call that function from within r_core_anal_search_xrefs and fcn_recurse.

**Comment by radare** [comment]
uhm yeah that seems right, but not sure how much shared code will those two implementations have, didnt checked the code yet. try it in a pr :)
